ntp-proto/src/packet.rs

```diff
@@ -113,25 +114,162 @@ pub struct NtpPacket<'a> {
     mac: Option<Mac<'a>>,
 }
 
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Clone, PartialEq, Eq)]
 pub enum ExtensionField<'a> {
-    Unknown { typeid: u16, data: Cow<'a, [u8]> },
+    UniqueIdentifier(Cow<'a, [u8]>),
+    NtsCookie(Cow<'a, [u8]>),
+    NtsCookiePlaceholder {
+        body_length: u16,
+    },
+    NtsEncryptedField {
+        nonce: Cow<'a, [u8]>,
+        plaintext: Cow<'a, [u8]>,
+    },
```

Comment:
Do we want to keep this in our datastructures as an explicit extension field? An NTS packet logically should never have more than one of these, and its existence essentially creates just a split in the extension fields (into open but signed, open and unsiged (if occuring after the nts encryption field) and encrypted (the plaintext in this design). It may be a better option to just handle that split on the main packet datastructure.

Reply:
well, it has the shape of an extension field, so we at least need to parse it as one. But we could separate that from the data that we actually present to the daemon layer.

Just to clarify, the only use that we foresee for this field is its current use or similar: the "plaintext" is the message so far? there is never some custom data that we put in there?

Reply:
No, the plaintext shouldn't be the message so far, or at least that is not what I assumed it would be. The plaintext in the crypto are the "encrypted" extension fields, (i.e. the ones that are not readable by third parties). As I read the spec, there are essentially 3 types of extension field in an nts packet. The validated ones (send in cleartext before the encryption ef), the encrypted ones (in the cyphertext of the encryption ef) and the unvalidated ones (send after the encryption ef). My suggestion here is to make that distinction explicit at the packet level.